### DISCUSSION TITLE
How does one automatically set the Test Details?

### DISCUSSION THREAD

**Discussion by SabreRunner** [fastlane/fastlane#22031] [General]
This is the only thing currently preventing me from a full automated build-to-testflight procedure.
What do I need to give Pilot so I won't have to go there and type it out manually?
![image](https://github.com/fastlane/fastlane/assets/10673434/14246fb2-c262-4435-a120-fa6e9feba8ba)

**Comment by mikestam35** [comment]
Hello, @SabreRunner [Receiptify Music](https://www.receiptify.net)


To automate your build-to-TestFlight process using Pilot, follow these steps:

Set Up Fastlane:

Make sure Fastlane is installed and configured in your project.
Create a Fastlane Configuration:

In your Fastfile, set up a lane for deployment that includes building the app and uploading to TestFlight.
Store Credentials Securely:

Use environment variables to store your Apple ID, password, and app-specific password (if using two-factor authentication).
Integrate with CI/CD:

Add Fastlane to your CI/CD pipeline (like GitHub Actions or Jenkins) to automate the process.

Summary

Ensure Fastlane is installed.
Configure a lane in Fastlane to handle build and TestFlight upload.
Store your Apple credentials securely using environment variables.
Add Fastlane commands to your CI/CD pipeline to automate the process.
This will allow you to automate the entire build-to-TestFlight process without needing manual intervention.

I hope it will be help you 

Best regards 
mike

**Reply by SabreRunner** [reply]
No, it doesn't. You didn't say anything about my original question and just spouted what looks like the most bland answer. Are you a bot?

**Comment by spencertransier** [comment]
@SabreRunner You can use the `changelog` parameter for `pilot`/`upload_to_testflight` to set the Test Details field. https://docs.fastlane.tools/actions/upload_to_testflight/ has a few examples.
